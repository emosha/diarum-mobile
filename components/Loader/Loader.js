import React from 'react';
import PropTypes from 'prop-types';
import { ActivityIndicator, View } from 'react-native';
import styles from './styles';

export default class Loader extends React.Component {
  static propTypes = {
    size: PropTypes.string,
    color: PropTypes.string,
  }

  render() {
    return (
      <View style={[styles.container, styles.horizontal]}>
        <ActivityIndicator size={this.props.size} color={this.props.color} />
      </View>
    );
  }
}
