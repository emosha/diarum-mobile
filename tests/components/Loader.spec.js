import React from 'react';
import Loader from '../../components/Loader/Loader';

test('renders correctly', () => {
  const tree = mount(<Loader size="40" color="red" />);
  expect(tree).toMatchSnapshot();
});
