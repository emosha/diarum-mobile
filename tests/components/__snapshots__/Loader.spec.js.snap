// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Loader
  color="red"
  size="40"
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "padding": 10,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "padding": 10,
          },
        ]
      }
    >
      <Component
        animating={true}
        color="red"
        hidesWhenStopped={true}
        size="40"
      >
        <ActivityIndicator
          animating={true}
          color="red"
          hidesWhenStopped={true}
          size="40"
        />
      </Component>
    </View>
  </Component>
</Loader>
`;
